Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -6,16 +6,22 @@ class BTAnalyticsService: Equatable {
 
     /// The HTTP client for communication with the analytics service endpoint. Exposed for testing.
     var http: BTHTTP?
-    
+
     /// The FPTI URL to post all analytic events.
     static let url = URL(string: "https://api-m.paypal.com")!
 
+    // MARK: - Private Properties
+
     private let apiClient: BTAPIClient
+    private let timerInterval: Double
+
+    private var events: [FPTIBatchData.Event] = []
```

Comment:
This queue of events is wiped if a merchant re-instantiates a new BTAPIClient. We should support the case where they create and destroy BTAPIClients on button tap

Reply:
I think this can be solved w/ a static property, it will be universally accessible to all BTAPIClient instances. However, arrays in Swift aren't thread-safe by default. So we'll want either a light array wrapper to ensure thread safety (like a DispatchQueue wrapper) _or_ explore another data store option for maintaining our events list that _is_ thread safe by default.